### ISSUE TITLE
Feeback on beautiful app and compile errors

### ISSUE THREAD

**Issue by estefan3112** [EmeraldLoc/sm_osx#25]
Hi, I stumbled over your repo - such a beautiful app on MacOS. Sincerely hope that you can manage to maintain it ... because currently compiles don't work for me (except for the coop-dx mod). Here some feedback:

1. I am on Apple Silicon and don't feel very comfy to have a parallel Intel64 Homebrew installed. Is that really necessary? I would suggest an optional Intel64 Homebrew install instead, as I fear a lot of issues with the two parallel Homebrews...
2. The "Configure Custom Repo" option does not work for me, the Next Button is greyed out, even if I fill out all the options. This option would indeed be useful to identify config issues in the pre-configured repos.
3. Because: Right now, I manage to compile sm64ex and Render96ex, which your app currently fails. With the custom repo option, I could try to identify the problem. 
4. All other repos, e.g. Moonshine, also fail to compile standalone with the same error, so that does not seem a problem of this app.

So again, many thanks and truly hope that you manage to keep working on this app! Cheers.

**Comment by EmeraldLoc** [comment]
> I am on Apple Silicon and don't feel very comfy to have a parallel Intel64 Homebrew installed. Is that really necessary? I would suggest an optional Intel64 Homebrew install instead, as I fear a lot of issues with the two parallel Homebrews...

It is necessary for some repos, so I have it install it automatically, this isn't necessary for a lot of repos, but for a lot of repos it is necessary. I could make this optional, and probably should.

> The "Configure Custom Repo" option does not work for me, the Next Button is greyed out, even if I fill out all the options

Do you have `.git` at the end of the clone url? I should make this more clear, or add this automatically.

> Right now, I manage to compile sm64ex and Render96ex, which your app currently fails. With the custom repo option, I could try to identify the problem.

So I used custom repos for these because the main repos were stale, since then they have both gotten updates.I haven't maintained these repos in 8 months, so I have no idea what's happened on them. I will look into converting these to the main R96 and the sm64ex repos if I haven't already (although I suspect I already did). I did get an issue related to libaudiofile, but that looked outside of my control, so hopefully it isn't related to that. And have you tried compiling at a speed slower than the fastest? I recall that fixing one of the repos compilation (just "Fast" will do the trick, need to revamp that system as well)

> All other repos, e.g. Moonshine, also fail to compile standalone with the same error, so that does not seem a problem of this app.

So every repo except coopdx, funny.

Ever since sm64coopdx began getting its precompiled builds (which is the main repo I run and maintain for macOS), I haven't had a need to launch sm_osx. I'm busy with a decent bit of stuff related to sm64coopdx modding, however if I find time, I'll try to remember to get an update for this, as one is warranted by now. I appreciate the issue here, and I'll try to look into it today if I find time :)

**Comment by estefan3112** [comment]
Ah, now all fine - the .git was missing indeed! So I can report you of an immediate success of building sm64ex2 with the following custom settings (Mac Mini M2 Pro):

Clone URL: https://github.com/sm64pc/sm64ex.git
Branch: nightly
Build Flags: BETTERCAMERA=1 NODRAWINGINSTANCE=1 TEXTURE_FIX=1 EXT_OPTIONS_MENU=1 EXTERNAL_DATA=1 -j10
And of course the OSX_BUILD=1 Flag
With this Custom Repo setting it builds very well!

**Comment by EmeraldLoc** [comment]
Alright, gotcha. I am currently in the process of revamping the repo system so it's all under one system, rather than all of it being hardcoded, which should make it easier to debug. Thanks!

**Comment by EmeraldLoc** [comment]
Figured I'd give an update:

I was able to get moonshine compiled, however it had major vertex errors, and crashed when entering stage 1, having it unplayable. Unsure of what happened, can only assume a gcc update broke it, have nothing else to go off of, so will probably be removing, especially since you can get this in sm64coopdx

I got sm64ex compiling easily, however I'm unsure why ex-alo is broken, I've never seen the error before so it will take time to fix.

sm64coopdx worked first try as expected, no issues there.

sm64ex-coop has been removed due to it's successor sm64coopdx.

The revamp for repos has been complete, I just need to make it work with patches now.

**Comment by estefan3112** [comment]
Another strange thing: sm64ex no longer runs with 60 fps, also if I build it directly from github in the Terminal. I have a functioning executable from 10 November that runs 60 fps, but now (probably after some homebrew updates), I no longer see 30 fps. Same with you? Thanks!

**Comment by estefan3112** [comment]
Ok, now I got it (however I did not remember that I did not have to apply the 60fps patch manually - back on track).

**Comment by estefan3112** [comment]
Plus: As soon as your new version is ready, I will add it to my recent blog post on SM64 on Silicon Macs. The launcher is so perfect for completing the picture!

https://mameonmacs.blogspot.com/2024/11/super-mario-64-hacks-on-apple-silicon.html

**Comment by estefan3112** [comment]
Hi, many thanks for the update, much appreciated! Some feedback, if I may:
1. sm_osx.app: builds nicely, but only if I add an empty folder called "Preview Content" into the directory.
2. render96ex: seems to build an older version of Render96, as the latest commits (rumble and highscreen fix) are missing.
3. sm64_ex_alo: has never built for me and still does not build
     Assembling: sound/sound_data.s -> build/us_pc/sound/sound_data.o
     clang: error: no input files
     gmake: *** [Makefile:1661: build/us_pc/sound/sound_data.o] Error 1
Many thanks for this app!

**Comment by EmeraldLoc** [comment]
Render96ex seems to be the latest build. The testerrt64 branch is being used, so don’t know where you’re seeing this. Probably won’t have to update the app, will add preview content folder onto GitHub.

**Comment by estefan3112** [comment]
Right, I overlooked - there is a MacOS-specific fork of Render96 that runs beautifully with rumble and Hires: https://github.com/flatmapthatshit/Render96ex_macOS

**Comment by EmeraldLoc** [comment]
Gotcha. Probably gonna stick with forking the official branch, but you can always add a custom repo and that’ll do the trick.
